lib/gman_client.rb

```diff
@@ -69,9 +78,16 @@ def retrieve_token
       JSON.parse(response)['access_token']
     end
 
+    private
+
     def convert_payload(response)
-      response.payload.first.nil? ? [response.to_h] : response.map(&:to_h)
+      response.map do |hash_request|
+        Hashie.symbolize_keys!(hash_request.to_h)
+      end
+    end
+
+    def attempt(retry_count)
```

Comment:
I prefer this, how do you feel about it?

I didn't mean to steam roll over the existing implementation (which in reflection I can see is how may previous comment would be taken), so I'm sorry about that.

I'm quite open to a discussion about the other approach if you want. :smile: